src/ae.c

```diff
@@ -378,22 +378,15 @@ int aeProcessEvents(aeEventLoop *eventLoop, int flags)
             tv.tv_sec = usUntilTimer / 1000000;
             tv.tv_usec = usUntilTimer % 1000000;
             tvp = &tv;
-        } else {
+        } else if ((flags & AE_DONT_WAIT) || (eventLoop->flags & AE_DONT_WAIT)) {
```

Comment:
there's a change in behavior here.
in the past, if `eventLoop->flags` has `AE_DONT_WAIT`, we would not wait, regardless of `usUntilTimer`
now, this condition is inside the `else`.
that can cause a bug in case of connTypeHasPendingData (for TLS).

Reply:
ok, I want to understand what you mean.
In fact, ```eventLoop->flags``` may be changed inside of the ```eventLoop->beforesleep```.
So we need to check ```eventLoop->flags``` after calling ```eventLoop->beforesleep```. If it does change, we may need to update the timeout (of course, this operation also depends on the parameter ```flags```).

Reply:
yes. it's a little bit ugly code, but the fact is that `eventLoop->flags` overrides the `flags` argument and instructs the next sleep to be skipped (ignoring timers sleep)

Reply:
ohh, you mean that the check for `if (eventLoop->flags & AE_DONT_WAIT) {` should be done after calling `eventLoop->beforesleep()`?
seems like you're right.
wanna re-purpose this PR?

Reply:
Yes, I will fix the above issue later in this PR.